src/diffusers/pipelines/kandinsky5/pipeline_kandinsky.py

```diff
@@ -0,0 +1,889 @@
+        max_sequence_length: int = 256,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        """
+        Encode prompt using Qwen2.5-VL text encoder.
+        
+        This method processes the input prompt through the Qwen2.5-VL model to generate
+        text embeddings suitable for video generation.
+        
+        Args:
+            prompt (Union[str, List[str]]): Input prompt or list of prompts
+            device (torch.device): Device to run encoding on
+            num_videos_per_prompt (int): Number of videos to generate per prompt
+            max_sequence_length (int): Maximum sequence length for tokenization
+            dtype (torch.dtype): Data type for embeddings
+            
+        Returns:
+            Tuple[torch.Tensor, torch.Tensor]: Text embeddings and cumulative sequence lengths
+        """
+        device = device or self._execution_device
+        dtype = dtype or self.text_encoder.dtype
+        prompt = [prompt] if isinstance(prompt, str) else prompt
+        prompt = [prompt_clean(p) for p in prompt]
+
+        # Kandinsky specific prompt template for detailed video description
+        prompt_template = "\n".join([
+            "<|im_start|>system\nYou are a promt engineer. Describe the video in detail.",
+            "Describe how the camera moves or shakes, describe the zoom and view angle, whether it follows the objects.",
+            "Describe the location of the video, main characters or objects and their action.",
+            "Describe the dynamism of the video and presented actions.",
+            "Name the visual style of the video: whether it is a professional footage, user generated content, some kind of animation, video game or scren content.",
+            "Describe the visual effects, postprocessing and transitions if they are presented in the video.",
+            "Pay attention to the order of key actions shown in the scene.<|im_end|>",
+            "<|im_start|>user\n{}<|im_end|>",
+        ])
+        crop_start = 129  # Position to start cropping from (system prompt tokens)
+        
+        full_texts = [prompt_template.format(p) for p in prompt]
+        
+        inputs = self.tokenizer(
+            text=full_texts,
+            images=None,
+            videos=None,
+            max_length=max_sequence_length + crop_start,
+            truncation=True,
+            return_tensors="pt",
+            padding=True,
+        ).to(device)
+
+        with torch.no_grad():
+            embeds = self.text_encoder(
+                input_ids=inputs["input_ids"],
+                return_dict=True,
+                output_hidden_states=True,
+            )["hidden_states"][-1][:, crop_start:]
+        
+        batch_size = len(prompt)
+        
+        attention_mask = inputs["attention_mask"][:, crop_start:]
+        cu_seqlens = torch.cumsum(attention_mask.sum(1), dim=0)
+        cu_seqlens = torch.cat([torch.zeros_like(cu_seqlens)[:1], cu_seqlens]).to(dtype=torch.int32)
```

Comment:
can we do something more staightforward? e.g. 
```suggestion
        cu_seqlens =F.pad(cu_seqlens, (1, 0), value=0)).to(dtype=torch.int32)
```